api/http/handler/endpoint.go

```diff
@@ -138,6 +140,15 @@ func (handler *EndpointHandler) handlePostEndpoints(w http.ResponseWriter, r *ht
 		AuthorizedTeams: []portainer.TeamID{},
 	}
 
+	if req.Local && req.URL == "" {
```

Comment:
There is no need for that extra `req.URL == ""` check. If the `Local` parameter is specified, it should automatically create a local endpoint and ignore the URL.

Reply:
@deviantony what about someone that wants has configured Docker to listen on a non-standard socket file or named pipe?

Reply:
Well you can always map it inside the Portainer container using `path/to/socket.go:/var/run/docker.sock`

When using Portainer as a binary it would then become a requirement, but this can still be an evolution that we can add later if required.

Reply:
fixed

Reply:
Mind updating the `api/swagger.yml` file as well ?

Reply:
Sorry, you mentioned already and I forgot - fixed.

Reply:
Awesome, thanks :-)